### DISCUSSION TITLE
How will Terragrunt be impacted by Terraform adopting BSL?

### DISCUSSION THREAD

**Discussion by dkujawski** [gruntwork-io/knowledge-base#759] [Help] [answered]
Hasicorp announced that they are moving their core applications over to the BSL:

- https://www.hashicorp.com/blog/hashicorp-adopts-business-source-license
- https://spdx.org/licenses/BUSL-1.1.html

How will this impact Terragrunt moving forward?  
We are exploring Terragrunt as a potential component in the future of our IaC pipelines.  What type of relationship, if any, does Gruntwork have with Hashicorp? 

---

<ins datetime="2023-08-10T20:52:07Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110391">Tracked in ticket #110391</a></p>
</ins>

**Comment by josh-padnick** [comment, marked as answer]
Gruntwork co-founder here. We just heard about this as well, and we'll begin the process of reviewing the implications immediately.

What's a little frustrating about this announcement is that they've included a blog post, discussion video, and press release, but they haven't published the actual BSL 1.1 license itself, which appears to be customized by each vendor to allow whatever "Additional Use Grants" they're comfortable with. So, we're going to need to get more information.

All that being said, my understanding is that Terragrunt is non-commercial use. Terragrunt is free and open sourced under the most permissive possible license (MIT). Furthermore, it contributes back to the Terraform community by effectively serving as an extension of Terraform to allow using Terraform at scale.

We'll be sure to post an update here when we have one.

**Update:** We've published an initial blog post on this at https://blog.gruntwork.io/the-impact-of-the-hashicorp-license-change-on-gruntwork-customers-5fcd096ba86a. We'll be covering the path forward beyond Terraform v1.5.5 in a separate blog post later this week.

**Reply by dkujawski** [reply]
Thank you for the quick response.

**Reply by estebanz01** [reply]
Here's a push of the new license to terraform: https://github.com/hashicorp/terraform/blob/e94ce2dd3877096b945309e38f3321b921f95b56/LICENSE

**Reply by josh-padnick** [reply]
Thanks. It looks like they're using almost an exact copy of the [MariaDB license](https://mariadb.com/bsl11/) with the most significant change being the following Additional Use Grant:

> Additional Use Grant: You may make production use of the Licensed Work,
provided such use does not include offering the Licensed Work
to third parties on a hosted or embedded basis which is
competitive with HashiCorp's products.

We'll post back here once we have an update.

**Reply by josh-padnick** [reply]
Update: We're still researching the implications for Terragrunt wrapping the newly licensed Terraform version going forward, but in the meantime, we're also exploring a novel idea: [Should Terragrunt be the open source version of Terraform going forward](https://github.com/gruntwork-io/terragrunt/issues/2662)?

**Reply by josh-padnick** [reply]
Update: We've published an initial blog post on this at https://blog.gruntwork.io/the-impact-of-the-hashicorp-license-change-on-gruntwork-customers-5fcd096ba86a. We'll be covering the path forward beyond Terraform v1.5.5 in a separate blog post later this week.

**Reply by josh-padnick** [reply]
**Update: Gruntwork supports a vendor-neutral, community-driven, truly open source Terraform, which we are pledging support to as part of [OpenTF](https://opentf.org/).** 

Going forward, Terragrunt will support OpenTF. If HashiCorp reverts their license change, we'll re-evaluate the best approach for the Terragrunt community.

See also our recent blog post: [The future of Terraform must be open](https://blog.gruntwork.io/the-future-of-terraform-must-be-open-ab0b9ba65bca)
